src/Compilers/CSharp/Portable/CodeGen/CodeGenerator.cs

```diff
@@ -133,7 +167,7 @@ private LocalDefinition LazyReturnTemp
                             constraints: LocalSlotConstraints.None,
                             isDynamic: false,
                             dynamicTransformFlags: ImmutableArray<TypedConstant>.Empty,
-                            isSlotReusable: localSymbol.SynthesizedKind.IsSlotReusable(_optimizations));
+                            isSlotReusable: _ilEmitStyle == ILEmitStyle.Release);
```

Comment:
Good point. 
I think we would be unable to reuse return variable anyways since it is scoped to the whole method.